### DISCUSSION TITLE
Is it possible to create a fix that creates another file?

### DISCUSSION THREAD

**Discussion by cdignam-segment** [eslint/eslint#17508] [Q&A] [answered]
I want to create an fixable rule that will generate a separate file? Is there a way to do this within the Eslint fixer api?

**Comment by Rec0iL99** [comment]
Hi @cdignam-segment, you could use the `fs` module from `nodejs` to create a file.

```js
fix(fixer) {
    fs.writeFile('hello.txt', "hello world", function (err) {
        if (err) throw err;
    })
},
```

**Reply by cdignam-segment** [reply]
Wouldn't that run even if `--fix` isn't provided though?

**Reply by Rec0iL99** [reply]
Ahh, sorry yes. I'm not sure if creating a new file in the `fix` function is recommended.

cc @nzakas

**Comment by nzakas** [comment, marked as answer]
No, there is no official way to do this. Fixes are made in the file in which the error was found.

**Comment by Fleshgrinder** [comment]
Actually, it is, but with two rules and a caveat. One rule checks the target file and generates it if its outdated (with `fixer.replaceTextRange(0, context.sourceCode.text.length, text)`). The other rule checks the source file and fails with a message that explains how to generate it for the first time if the target is missing. This won't result in full automation (sadly), but might be good enough. Especially if the target file is checked in and needs to be generated only once to get started.

An alternative approach that works is to check for `process.argv.includes('--fix')` and generate the file on the first visit.
